internal/provider/resource_managed_identity_alias.go

```diff
@@ -0,0 +1,306 @@
+		ID: state.ID.ValueString(),
+	})
+	if err != nil {
+		if tharsis.NotFoundError(err) {
+			resp.State.RemoveResource(ctx)
+			return
+		}
+
+		resp.Diagnostics.AddError(
+			"Error reading managed identity alias",
+			err.Error(),
+		)
+		return
+	}
+
+	// Copy the from-Tharsis struct to the state.
+	if err = t.copyManagedIdentityAlias(*found, &state); err != nil {
+		resp.Diagnostics.AddError(
+			"Error setting state",
+			err.Error(),
+		)
+		return
+	}
+
+	// Set the refreshed state, whether or not there is an error.
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (t *managedIdentityAliasResource) Update(ctx context.Context,
+	req resource.UpdateRequest, resp *resource.UpdateResponse) {
+
+	// Retrieve values from plan.
+	var plan ManagedIdentityModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var state ManagedIdentityModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if !reflect.DeepEqual(plan, state) {
+		resp.Diagnostics.AddError(
+			"Error updating managed identity alias",
+			"An alias cannot be updated",
+		)
+	}
+}
+
+func (t *managedIdentityAliasResource) Delete(ctx context.Context,
+	req resource.DeleteRequest, resp *resource.DeleteResponse) {
+
+	// Get the current state.
+	var state ManagedIdentityAliasModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Delete the managed identity alias via Tharsis.
+	// The ID is used to find the record to delete.
+	err := t.client.ManagedIdentity.DeleteManagedIdentityAlias(ctx,
+		&ttypes.DeleteManagedIdentityAliasInput{
+			ID: state.ID.ValueString(),
+		})
+	if err != nil {
+
+		// Handle the case that the managed identity alias no longer exists.
+		if tharsis.NotFoundError(err) {
+			resp.State.RemoveResource(ctx)
+			return
+		}
+
+		resp.Diagnostics.AddError(
+			"Error deleting managed identity alias",
+			err.Error(),
+		)
+	}
+}
+
+// ImportState helps the provider implement the ResourceWithImportState interface.
+func (t *managedIdentityAliasResource) ImportState(ctx context.Context,
+	req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+
+	// Retrieve import ID and save to id attribute
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+// copyManagedIdentityAlias copies the contents of a managed identity alias.
+// It is intended to copy from a struct returned by Tharsis to a Terraform plan or state.
+func (t *managedIdentityAliasResource) copyManagedIdentityAlias(src ttypes.ManagedIdentity, dest *ManagedIdentityAliasModel) error {
+
+	dest.ID = types.StringValue(src.Metadata.ID)
+	dest.ResourcePath = types.StringValue(src.ResourcePath)
+	dest.Name = types.StringValue(src.Name)
+	dest.GroupPath = types.StringValue(t.getGroupPath(src.ResourcePath))
+
+	if src.AliasSourceID != nil {
```

Comment:
Will `AliasSourceID` ever be nil? Seems like the SDK should always be returning this field for aliases.

Reply:
AliasSourceID should never be null for a managed identity alias.